test/e2e/network_segmentation.go

```diff
@@ -46,6 +46,10 @@ var _ = Describe("Network Segmentation", func() {
 		nadClient, err = nadclient.NewForConfig(f.ClientConfig())
 		Expect(err).NotTo(HaveOccurred())
 	})
+	AfterEach(func() {
```

Comment:
Why is it necessary to delete the pods although namespace is deleted when test is done?

Reply:
The networks are only cleaned up properly if all the pods on a network are removed before the network itself, otherwise artifacts are left in the database. In the future this can be removed when it is decided what pods should do when there network is removed.

Reply:
Make sense, thanks for explaining

Reply:
Does it make sense to also clean up pod in this test? https://github.com/ovn-org/ovn-kubernetes/blob/6859d3dc4e823de8bce7fe36fceaccf54ed09710/test/e2e/network_segmentation.go#L439

Reply:
yes, added the cleanup